cardano-api/src/Cardano/Api/Modes.hs

```diff
@@ -132,14 +132,16 @@ deriving instance Show (ConsensusModeIsMultiEra mode)
 
 toEraInMode :: CardanoEra era -> ConsensusMode mode -> Maybe (EraInMode era mode)
 toEraInMode ByronEra   ByronMode   = Just ByronEraInByronMode
+toEraInMode _          ByronMode   = Nothing
 toEraInMode ShelleyEra ShelleyMode = Just ShelleyEraInShelleyMode
+toEraInMode _          ShelleyMode = Nothing
 toEraInMode ByronEra   CardanoMode = Just ByronEraInCardanoMode
 toEraInMode ShelleyEra CardanoMode = Just ShelleyEraInCardanoMode
 toEraInMode AllegraEra CardanoMode = Just AllegraEraInCardanoMode
 toEraInMode MaryEra    CardanoMode = Just MaryEraInCardanoMode
 toEraInMode AlonzoEra  CardanoMode = Just AlonzoEraInCardanoMode
 toEraInMode BabbageEra CardanoMode = Just BabbageEraInCardanoMode
-toEraInMode _ _                    = Nothing
+toEraInMode ConwayEra  CardanoMode = Just ConwayEraInCardanoMode
```

Comment:
I'm wondering if it would make sense to add a property test to catch similar bugs in the future.

Reply:
The type system should be able to catch this.  By removing `_`, not adding new eras in the pattern match will make it in-exhaustive so we'll get an error.

Reply:
Assuming that the mapping is correct in the first place 😃